### PR TITLE
feat: remote write proxy support

### PR DESCRIPTION
**Pull request by pb82**
Support the Observability Proxy and don't create the token-refresher if the proxy config secret exists.

### CODE REVIEW

File: controllers/reconcilers/configuration/configuration_reconciler.go [hunk 1]
```diff
@@ -793,3 +793,49 @@ func (r *Reconciler) waitForResourcesDeployment(ctx context.Context, cr *v1.Obse
...
+		}
+
+		return nil, err
+	}
+
+	return secret, nil
+}
+
+func (r *Reconciler) GetRemoteWriteOauth2Config(ctx context.Context, cr *v1.Observability) (*prometheusv1.OAuth2, error) {
+	secret, err := r.GetOauth2ConfigSecret(ctx, cr)
```
Comment (vmanley) [inline controllers/reconcilers/configuration/configuration_reconciler.go:817]: Should we return nil config if the secret is nil?

Comment (pb82) [reply controllers/reconcilers/configuration/configuration_reconciler.go:817]: Yes, good point

File: controllers/reconcilers/configuration/configuration_reconciler.go [hunk 2]
```diff
@@ -793,3 +793,53 @@ func (r *Reconciler) waitForResourcesDeployment(ctx context.Context, cr *v1.Obse
...
+				Key: "client_id",
+			},
+		},
+		ClientSecret: corev1.SecretKeySelector{
+			LocalObjectReference: corev1.LocalObjectReference{
+				Name: RemoteWriteOIDCSecretName,
+			},
+			Key: "client_secret",
+		},
+		TokenURL: string(secret.Data["token_url"]),
```
Comment (vmanley) [inline controllers/reconcilers/configuration/configuration_reconciler.go:841]: Not 100% sure on the key from the secret here. Should it be "issuer_url"?

[the remark this answers is not in the document]
Comment (pb82) [reply controllers/reconcilers/configuration/configuration_reconciler.go:841]: Yes, you and @vmanley are correct. It should be issuer_url

[the remark this answers is not in the document]
Comment (machi1990) [reply controllers/reconcilers/configuration/configuration_reconciler.go]: @pb82 please the other part of my comment regarding the missing `/protocol/openid-connect/token` in the url. i.e I suspect that prometheus requires a token url, while we are just providing an issuer url. So we'll need to append `/protocol/openid-connect/token` from the value

File: controllers/reconcilers/configuration/configuration_reconciler.go [hunk 3]
```diff
@@ -793,3 +793,53 @@ func (r *Reconciler) waitForResourcesDeployment(ctx context.Context, cr *v1.Obse
...
+				Key: "client_id",
+			},
+		},
+		ClientSecret: corev1.SecretKeySelector{
+			LocalObjectReference: corev1.LocalObjectReference{
+				Name: RemoteWriteOIDCSecretName,
+			},
+			Key: "client_secret",
+		},
+		TokenURL: string(secret.Data["issuer_url"]),
```
Comment (machi1990) [inline controllers/reconcilers/configuration/configuration_reconciler.go:841]: please see https://github.com/redhat-developer/observability-operator/pull/194#discussion_r1129696376

**Review by machi1990 — APPROVED** [review]
the code change lgtm

I've not however verified it.

**Review by vmanley — APPROVED** [review]

**Pull request merged**